### PR TITLE
refactor(addon-commerce): `InputCard` migrate to new textfield

### PR DESCRIPTION
**Pull request by waterplea**
Closes #

### CODE REVIEW

**Review by nsbarsukov — APPROVED** [review]
Previous implementation of `CardNumber` did not add spaces in form control value.
New version does.
Is it expected change ?

**Review by splincode — APPROVED** [review]

File: projects/addon-commerce/components/input-card/input-card.directive.ts [hunk 1]
```diff
@@ -0,0 +1,74 @@
...
+        this.mask.options = {mask: TUI_CARD_MASK};
+        this.mask.ngOnChanges();
+
+        if (!this.accessor) {
+            return;
+        }
+
+        const onChanges = this.accessor.onChange.bind(this.accessor);
+
+        this.accessor.onChange = (value: string) => onChanges(value.replaceAll(' ', ''));
```
Comment (nsbarsukov) [inline projects/addon-commerce/components/input-card/input-card.directive.ts:60]: What is about vice-versa operation (control => textfield value; `writeValue`) ?

```ts
import {Component} from '@angular/core';
import {FormControl, ReactiveFormsModule} from '@angular/forms';
import {TuiInputCard} from '@taiga-ui/addon-commerce';
import {TuiTextfield} from '@taiga-ui/core';

@Component({
    standalone: true,
    imports: [ReactiveFormsModule, TuiTextfield, TuiInputCard],
    template: `
        <tui-textfield>
            <input
                tuiInputCard
                tuiTextfield
                [formControl]="control"
            />
        </tui-textfield>
    `,
})
export default class ExampleComponent {
    protected control = new FormControl('12345678');
}
```

Comment (waterplea) [reply projects/addon-commerce/components/input-card/input-card.directive.ts:60]: I believe that is taken care of by Maskito directive

File: projects/addon-commerce/components/input-card-grouped/abstract-input-card.ts [hunk 2]
```diff
@@ -1,12 +1,11 @@
 import {Directive, EventEmitter, inject, Input, Output} from '@angular/core';
+import type {TuiInputCardOptions} from '@taiga-ui/addon-commerce/components/input-card/input-card.options';
```
Comment (waterplea) [inline projects/addon-commerce/components/input-card-grouped/abstract-input-card.ts:2]:
```suggestion
import type {TuiInputCardOptions} from '@taiga-ui/addon-commerce/components/input-card';
```

**Pull request merged**